DNN Platform/Library/Entities/Portals/PortalController.cs

```diff
@@ -2815,19 +2815,24 @@ public static string DeletePortal(PortalInfo portal, string serverPath)
                         }
                     }
                     //delete upload directory
-                    Globals.DeleteFolderRecursive(serverPath + "Portals\\" + portal.PortalID);
                     if (!string.IsNullOrEmpty(portal.HomeDirectory))
                     {
-                        string homeDirectory = portal.HomeDirectoryMapPath;
-
-                        if (homeDirectory.EndsWith("\\"))
-                        {
-                            homeDirectory = homeDirectory.Substring(0, homeDirectory.Length - 1);
-                        }
-
-                        if (Directory.Exists(homeDirectory))
+                        var homeDirectory = portal.HomeDirectoryMapPath;
+                        // check whether home directory is not used by other portal
+                        // it happens when new portal creation failed, but home directory defined by user is alredy in use with other portal
+                        var homeDirectoryInUse = portals.OfType<PortalInfo>().Any(
+                            x => x.PortalID != portal.PortalID &&
+                            x.HomeDirectoryMapPath.Equals(homeDirectory, StringComparison.OrdinalIgnoreCase));
+                        if (!homeDirectoryInUse)
                         {
-                            Globals.DeleteFolderRecursive(homeDirectory);
+                            if (homeDirectory.EndsWith("\\"))
+                            {
+                                homeDirectory = homeDirectory.Substring(0, homeDirectory.Length - 1);

```

Comment:
It would be simpler to simply trim: `homeDirectory = homeDirectory.TrimEnd('\\')`. No need for condition in that case

Reply:
And by the way, why is it even necessary to remove the trailing backslash? Is it not all the same for `Directory.Exists`?